hypothesis-python/src/hypothesis/strategies/_internal/numbers.py

```diff
@@ -93,8 +94,15 @@ def do_draw(self, data):
         return d.integer_range(data, self.start, self.end, center=0)
 
     def filter(self, condition):
+        if condition is math.isfinite:
+            return self
+        if condition in [math.isinf, math.isnan]:
+            return nothing()
         kwargs, pred = get_integer_predicate_bounds(condition)
 
+        if pred == UNSATISFIABLE.predicate:
+            return nothing()
+
```

Comment:
I'm not a fan of this special case.  How about we change `UNSATISFIABLE` to be `ConstructivePredicate({"min_value": 1,"max_value": -1}, None})` instead?  Then it would be handled correctly by the bounds logic below.